server/storage/mvcc/watcher.go

```diff
@@ -123,11 +124,7 @@ func (ws *watchStream) Watch(id WatchID, key, end []byte, startRev int64, fcs ..
 	}
 
 	if id == clientv3.AutoWatchID {
-		for ws.watchers[ws.nextID] != nil {
```

Comment:
sounds good to use unique ID across all the watch streams.
it's draft version. just want to mention that it still needs to avoid allocate that ID which has been occupied by user request.